stl/inc/stop_token

```diff
@@ -0,0 +1,110 @@
+// stop_token standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _STOP_TOKEN_
+#define _STOP_TOKEN_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#if _HAS_CXX20
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+class stop_token {
+public:
+    stop_token() noexcept;
+
+    ~stop_token();
+
+    stop_token(stop_token&&) noexcept;
+    stop_token& operator=(stop_token&&) noexcept;
+
+    stop_token(const stop_token&) noexcept;
+    stop_token& operator=(const stop_token&) noexcept;

```

Comment:
Here and in `stop_source`, should you follow the Standard's ordering of copy ctor, move ctor, copy assign, move assign? It is purely stylistic, of course.

Reply:
Ahh yes. So I modeled the ordering of special member functions mostly based off of some other classes within Microsfot/STL, but I also personally prefer following the Standard's ordering.